nemo_skills/pipeline/utils/eval.py

```diff
@@ -28,6 +28,66 @@
 LOG = logging.getLogger(get_logger_name(__file__))
 
 
+def parse_eval_args(eval_args: str) -> tuple[str | None, dict]:
+    """Parse eval_args string to extract eval_type and eval_config.
+
+    Handles Hydra argument formats:
+    - ++eval_type=value (override)
+    - +eval_type=value (new)
+    - eval_type=value (config)
+    """
+    if not eval_args:
+        return None, {}
+
+    import re
+    import shlex
+
+    eval_type = None
+    eval_config = {}
+
+    # Parse eval_args to extract eval_type and eval_config
+    eval_arg_parts = shlex.split(eval_args)
+    for part in eval_arg_parts:
+        # Match eval_type with any Hydra prefix
+        eval_type_match = re.match(r"^(\+{0,2})eval_type=(.+)$", part)
+        if eval_type_match:
+            eval_type = eval_type_match.group(2)
+            continue
+
+        # Match eval_config with any Hydra prefix
+        eval_config_match = re.match(r"^(\+{0,2})eval_config\.(.+)$", part)
+        if eval_config_match:
+            config_part = eval_config_match.group(2)
+            if "=" in config_part:
+                key, value = config_part.split("=", 1)
+                # Handle nested keys like sandbox.timeout
+                if "." in key:
+                    main_key, sub_key = key.split(".", 1)
+                    if main_key not in eval_config:
+                        eval_config[main_key] = {}
+                    eval_config[main_key][sub_key] = value
+                else:
+                    eval_config[key] = value
+
+    return eval_type, eval_config
+
+
+def should_use_single_eval(eval_args: str) -> bool:
+    """Determine if evaluation should be done during generation (single) vs after (batch)."""
+    eval_type, eval_config = parse_eval_args(eval_args)
+
+    if not eval_type:
+        return False
+
+    try:
+        from nemo_skills.evaluation.evaluator import supports_single_eval
+
+        return supports_single_eval(eval_type, eval_config)
+    except Exception:
```

Comment:
let's not do this, just let it fail, it's at job submission time so best to just report errors and not hide them. And we generally don't expect this to fail, so it's abnormal